src/game_updater/GameUpdater.cpp

```diff
@@ -58,11 +58,20 @@ void GameUpdater::Init(std::shared_ptr<GameState> game_state) {
 }
 
 void GameUpdater::Update(std::shared_ptr<GameState> game_state) {
+  std::shared_ptr<sf::Music> music = game_state->GetLevel()->getMusic();
+
   // check to see if the music ended, in which case the level is complete
-  if (game_state->GetLevel()->getMusic()->getStatus() ==
-      sf::Music::Status::Stopped) {
+  if (music->getStatus() == sf::Music::Status::Stopped &&
+      music->getPlayingOffset() != sf::Time::Zero) {
```

Comment:
This looks a little confusing may I suggest
`((music->getStatus() == sf::Music::Status::Stopped) && (music->getPlayingOffset() != sf::Time::Zero))`
A little extra parenthesis never hurt